api/redhatopenshift/HcpCluster.Management/hcpCluster-models.tsp

```diff
@@ -486,3 +505,108 @@ model HcpOpenShiftClusterAdminCredential {
  *   End HCP cluster credentials
  * =======================================
  */
+
+/*
+ * =======================================
+ *  ExternalAuth resources
+ * =======================================
+ */
+
+/** External Auth profile */
+model ExternalAuthProperties {
+  /** Provisioning state */
+  @visibility(Lifecycle.Read)
+  provisioningState?: ExternalAuthProvisioningState;
+
+  /** Token Issuer profile */
+  issuer: TokenIssuerProfile;
+
+  /** External Auth clients */
+  @maxItems(20)
+  clients?: ExternalAuthClientProfile[];
+
+  /** External Auth claim */
+  claim: ExternalAuthClaimProfile;
+}
+
+/** Token issuer profile */
+model TokenIssuerProfile {
+  /** The URL of the token issuer */
+  url: url;
+
+  /** The audience of the token issuer */
+  @minItems(1)
+  audiences: string[];
+
+  /** The issuer of the token */
+  ca?: string;
+}
+
+/** External Auth client profile */
+model ExternalAuthClientProfile {
+  /** External Auth client component */
+  component?: ExternalAuthClientComponentProfile;
+
+  /** external Auth client id */
+  id: string;
+
+  /** external auth client scopes */
+  extraScopes?: string[];
+}
+
+/** External Auth component profile */
+model ExternalAuthClientComponentProfile {
+  /** The name of the external Auth client */
+  name?: string;
+
+  /** The namespace of the external Auth client */
+  authClientNamespace?: string;
+}
+
+/** External Auth claim profile */
+model ExternalAuthClaimProfile {
+  /** The claim mappings */
+  mappings: TokenClaimMappingsProfile;
+
+  /** The claim validation rules */
+  @OpenAPI.extension("x-ms-identifiers", #["claim", "requiredValue"])
+  validationRules?: TokenClaimValidationRuleProfile[];
+}
+
+/** External Auth claim mappings profile.
+ * At a minimum username or groups must be defined.
+ */
+model TokenClaimMappingsProfile {
```

Comment:
intentional mismatch to HCP with UID and Extra?

Reply:
Looks like these aren't supported atm: https://github.com/openshift-online/ocm-api-model/blob/12017b3afa4bbb985dc5ed4b1189f1497c480875/model/clusters_mgmt/v1/external_auth_type.model#L144-L160

@JameelB to confirm